interpreter/gpu/cuda.go

```diff
@@ -0,0 +1,152 @@
+package gpu // import "go.opentelemetry.io/ebpf-profiler/interpreter/gpu"
+
+import (
+	"runtime"
+
+	log "github.com/sirupsen/logrus"
+	"go.opentelemetry.io/ebpf-profiler/interpreter"
+	"go.opentelemetry.io/ebpf-profiler/libpf"
+	"go.opentelemetry.io/ebpf-profiler/libpf/pfelf"
+	"go.opentelemetry.io/ebpf-profiler/remotememory"
+)
+
+type data struct {
+	path   string
+	link   interpreter.LinkCloser
+	probes []pfelf.USDTProbe
+}
+
+type instance struct {
+	path string
+	interpreter.InstanceStubs
+	link interpreter.LinkCloser
+}
+
+func Loader(ebpf interpreter.EbpfHandler, info *interpreter.LoaderInfo) (interpreter.Data, error) {
+	ef, err := info.GetELF()
+	if err != nil {
+		return nil, err
+	}
+	// We use the existence of the .note.stapsdt section to determine if this is a
+	// process that has libparcagpucupti.so loaded. Its cheaper and more reliable than loading
+	// the symbol table.
+	if sec := ef.Section(".note.stapsdt"); sec != nil {
+		probes, err := pfelf.ParseUSDTProbes(sec)
+		if err != nil {
+			return nil, err
+		}
+		var parcagpuProbes []pfelf.USDTProbe
+		for _, probe := range probes {
+			if probe.Provider == "parcagpu" {
+				parcagpuProbes = append(parcagpuProbes, probe)
+			}
+		}
+		if len(parcagpuProbes) != 3 {
+			return nil, nil
+		}
+
+		// Validate probe arguments match what cuda.ebpf.c expects
+		validateProbeArguments(parcagpuProbes, info.FileName())
+
+		return &data{path: info.FileName(),
+			probes: parcagpuProbes}, nil
+	}
+	return nil, nil
+}
+
+// validateProbeArguments checks that the USDT probe arguments match the expectations
+// in cuda.ebpf.c and logs errors if they don't match.
+func validateProbeArguments(probes []pfelf.USDTProbe, path string) {
+	var expectedProbes map[string]string
```

Comment:
Should we return an error and refuse to attach the interpreter if this function fails?